index.js

```diff
@@ -31,7 +31,7 @@ async function getStream(inputStream, options) {
 			reject(error);
 		};
 
-		stream = pump(inputStream, bufferStream(options), error => {
+		pipeline(inputStream, stream, error => {
```

Comment:
You could `util.promisify` and await the pipeline.

Reply:
I would prefer not to make an promise executor an async function to be able to use await
https://eslint.org/docs/rules/no-async-promise-executor
So I used .then with the promise created by the promisified stream.pipeline function

Reply:
You don't have to do that though. You can put an async-IIFE inside the executer function with a try-catch inside.

Reply:
Like this?
```js
(async () => {
	try {
		await streamPipelinePromisified(inputStream, stream);
		resolve();
	} catch (error) {
		rejectPromise(error);
	}
})();
```

Reply:
Yes